docs/new_problem.md

```diff
@@ -15,9 +15,9 @@ Let's add a new dataset together and train the transformer model. We'll be learn
 
 For each problem we want to tackle we create a new problem class and register it. Let's call our problem `Word2def`.
 
-Since many text2text problems share similar methods, there's already a class called `Text2TextProblem` that extends the base problem class, `Problem` (both found in `problem.py`).
+Since many text2text problems share similar methods, there's already a class called [`Text2TextProblem`](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/data_generators/problem.py#L354) that extends the base problem class, `Problem` (both found in [`problem.py`](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/data_generators/problem.py)).
```

Comment:
https://github.com/tensorflow/tensor2tensor/blob/24071ba/tensor2tensor/data_generators/problem.py#L354

When referencing a particular line it is always better to provide permalink (including commit id, hint: press "y" in github web). Otherwise, the link becomes misleading once someone adds or deletes some lines in problem.py.

Reply:
That makes a lot of sense. I had followed what's being done in the main readme. Will update this one. Thanks for the feedback